routing/route/route.go

```diff
@@ -76,6 +77,10 @@ type Hop struct {
 	// carries as a fee will be subtracted by the hop.
 	AmtToForward lnwire.MilliSatoshi
 
+	// MPP encapsulates the data required for option_mpp. This field should
+	// only be set for the final hop.
+	MPP *record.MPP
```

Comment:
Does it need to be nested, or can those fields be top level fields? Probably dependent on what will happen when the set is extended to amp.

Reply:
IMO it's better nested (with a pointer) as otherwise it isn't clear what the required vs optional fields are in the `Hop` struct itself.